server/src/main/java/io/spine/server/integration/ThirdPartyContext.java

```diff
@@ -0,0 +1,190 @@
+ * Bounded Contexts and dispatched via {@link IntegrationBus}.
+ *
+ * @implSpec Note that a {@code ThirdPartyContext} sends a request for external messages to
+ *         other contexts. The {@code ThirdPartyContext} never consumes external messages itself,
+ *         but requires the other Bounded Contexts to send their requests, so that the publishing
+ *         channels are open. Depending of the implementation of
+ *         {@link io.spine.server.transport.TransportFactory transport}, creating
+ *         a {@code ThirdPartyContext} may be an expensive operation. Thus, it is recommended that
+ *         the instances of this class are reused and {@linkplain #close() closed} when they are
+ *         no longer needed.
+ */
+public final class ThirdPartyContext implements AutoCloseable {
+
+    private final BoundedContext context;
+    private final Any producerId;
+
+    /**
+     * Creates a new single-tenant instance of {@code ThirdPartyContext} with the given name.
+     *
+     * @param name
+     *         name of the third-party system
+     */
+    public static ThirdPartyContext singleTenant(String name) {
+        checkNotNull(name);
+        return newContext(name, false);
+    }
+
+    /**
+     * Creates a new multitenant instance of {@code ThirdPartyContext} with the given name.
+     *
+     * @param name
+     *         name of the third-party system
+     */
+    public static ThirdPartyContext multitenant(String name) {
+        checkNotNull(name);
+        return newContext(name, true);
+    }
+
+    private static ThirdPartyContext newContext(String name, boolean multitenant) {
+        BoundedContext context = notStoringEvents(name, multitenant).build();
+        context.integrationBus()
+               .notifyOfCurrentNeeds();
+        return new ThirdPartyContext(context);
+    }
+
+    private ThirdPartyContext(BoundedContext context) {
+        this.context = context;
+        this.producerId = pack(context.name());
+    }
+
+    /**
+     * Emits an event from the third-party system.
+     *
+     * <p>If the event is required by another Context, posts the event into
+     * the {@link IntegrationBus} of the respective Context. Does nothing if the event is not
+     * required by any Context.
+     *
+     * <p>The caller is required to supply the tenant ID via the {@code ActorContext.tenant_id} if
+     * this Context is multitenant.
+     *
+     * @param actorContext
+     *         the info about the actor, a user or a software component, who emits the event
+     * @param eventMessage
+     *         the event
+     */
+    public void emittedEvent(ActorContext actorContext, EventMessage eventMessage) {
+        checkNotNull(actorContext);
+        checkNotNull(eventMessage);
+        checkTenant(actorContext, eventMessage);
+
+        EventId id = EventId
+                .newBuilder()
+                .setValue(newUuid())
+                .build();
+        EventContext eventContext = EventContext
```

Comment:
Please have `EventFactory` to avoid generating event IDs and assembling `Event` instances. We need to encapsulate these details of creating events even.